src/Microsoft.Health.Fhir.Core/Features/Conformance/GetSmartConfigurationHandler.cs

```diff
@@ -5,67 +5,91 @@
 
 using System;
 using System.Collections.Generic;
+using System.Linq;
 using System.Net;
+using System.Net.Http;
+using System.Net.Http.Json;
 using System.Threading;
 using System.Threading.Tasks;
+using AngleSharp;
 using EnsureThat;
 using MediatR;
+using Microsoft.Build.Framework;
+using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
+using Microsoft.Health.Core.Configs;
 using Microsoft.Health.Fhir.Core.Configs;
+using Microsoft.Health.Fhir.Core.Features.Conformance.Models;
+using Microsoft.Health.Fhir.Core.Features.Conformance.Providers;
 using Microsoft.Health.Fhir.Core.Features.Operations;
 using Microsoft.Health.Fhir.Core.Messages.Get;
 using Microsoft.Health.Fhir.Core.Models;
+using Newtonsoft.Json;
 
 namespace Microsoft.Health.Fhir.Core.Features.Conformance
 {
     public class GetSmartConfigurationHandler : IRequestHandler<GetSmartConfigurationRequest, GetSmartConfigurationResponse>
     {
         private readonly SecurityConfiguration _securityConfiguration;
+        private readonly IWellKnownConfigurationProvider _configurationProvider;
+        private readonly ILogger<GetSmartConfigurationHandler> _logger;
 
-        public GetSmartConfigurationHandler(IOptions<SecurityConfiguration> securityConfigurationOptions)
+        public GetSmartConfigurationHandler(
+            IOptions<SecurityConfiguration> securityConfigurationOptions,
+            IWellKnownConfigurationProvider configurationProvider,
+            ILogger<GetSmartConfigurationHandler> logger)
         {
-            EnsureArg.IsNotNull(securityConfigurationOptions?.Value, nameof(securityConfigurationOptions));
-
-            _securityConfiguration = securityConfigurationOptions.Value;
-        }
-
-        public Task<GetSmartConfigurationResponse> Handle(GetSmartConfigurationRequest request, CancellationToken cancellationToken)
-        {
-            return Task.FromResult(Handle(request));
+            _securityConfiguration = EnsureArg.IsNotNull(securityConfigurationOptions?.Value, nameof(securityConfigurationOptions));
+            _configurationProvider = EnsureArg.IsNotNull(configurationProvider, nameof(configurationProvider));
+            _logger = EnsureArg.IsNotNull(logger, nameof(logger));
         }
 
-        protected GetSmartConfigurationResponse Handle(GetSmartConfigurationRequest request)
+        public async Task<GetSmartConfigurationResponse> Handle(GetSmartConfigurationRequest request, CancellationToken cancellationToken)
```

Comment:
It shouldn't. The SMART on FHIR proxy relied on the capability statement for discovery, and was never properly updated to use the .well-known/smart-configuration.  It looks like this code never actually worked because simply appending "/authorize" and "/token" to the authority does not work with most identity providers. The proper way to get the authorization and token endpoints is to call the well-known endpoint.

```c
Uri authorizationEndpoint = new Uri(baseEndpoint + "/authorize");
Uri tokenEndpoint = new Uri(baseEndpoint + "/token");
ICollection<string> capabilities = new List<string>
{
    "sso-openid-connect",
    "permission-offline",
    "permission-patient",
    "permission-user",
};
```